### PR TITLE
fixing spider for Loblaws

### PR DESCRIPTION
**Pull request by boomerwv1**

### CODE REVIEW

File: locations/spiders/loblaws.py [hunk 1]
```diff
@@ -11,94 +11,24 @@ class LoblawsSpider(scrapy.Spider):
     item_attributes = {"brand": "Loblaws"}
```
Comment (Cj-Malone) [inline locations/spiders/loblaws.py:11]:
```suggestion
    item_attributes = {"brand": "Loblaws", "brand_wikidata": "Q3257626"}
```

File: locations/spiders/loblaws.py [hunk 2]
```diff
@@ -11,94 +11,24 @@ class LoblawsSpider(scrapy.Spider):
...
-                )
-
-            yield GeojsonPointItem(**properties)
+        for i in results:
+            if i['locationType'] == 'STORE':
+                properties = {
+                    "ref": i["storeId"],
+                    "name": i["name"],
+                    "lat": i["geoPoint"]['latitude'],
+                    "lon": i["geoPoint"]['latitude'],
```
Comment (Cj-Malone) [inline locations/spiders/loblaws.py:26]:
```suggestion
                    "lon": i["geoPoint"]['longitude'],
```

**Review by Cj-Malone — CHANGES_REQUESTED** [review]
Looks good, but a little to do.

Do we need to check `storeBannerId` to see what type of store it is? (Are they all brand=Loblaws?)
| count | storeBannerId           |
|-------|-------------------------|
| 5     | "independentcitymarket" |
| 11    | "dominion"              |
| 14    | "rapid"                 |
| 30    | "extrafoods"            |
| 43    | "zehrs"                 |
| 44    | "fortinos"              |
| 52    | "rass"                  |
| 60    | "loblaw"                |
| 61    | "wholesaleclub"         |
| 76    | "valumart"              |
| 124   | "superstore"            |
| 125   | "maxi"                  |
| 162   | "provigo"               |
| 216   | "independent"           |
| 425   | "nofrills"              |

Can we add the website? It seems to be `https://www.loblaws.ca/store-locator/details/{ref}`

The black test is failing because it wants the file to be formatted in a specific way, it's not the end of the world, but if you want to fix it. You may need to install black with something like `pip install black` and then run it on the file `black locations/spiders/loblaws.py`

File: locations/spiders/loblaws.py [hunk 3]
```diff
@@ -11,94 +11,24 @@ class LoblawsSpider(scrapy.Spider):
...
+                    "addr_full": i["address"]["line1"],
```
Comment (Cj-Malone) [inline locations/spiders/loblaws.py:27]:
```suggestion
                    "street_address": ", ".join(
                        filter(
                            None, [i["address"].get("line2"), i["address"].get("line1")]
                        )
                    ),
```

**Comment by boomerwv1** [comment]
Good question. It looks like Loblaws if the parent company of all of these names. They are described as

`operating under 22 regional and market-segment banners`

Loblaws is one of these banners with approx 60 locations. I'm open to suggestions on if we want to include all of the banners in this spider or just the specific Loblaws branded ones.

**Comment by Cj-Malone** [comment]
I think yes, we want to harvest this data in this spider. If we had a separate spider for them we wouldn't want to duplicate the data, but since we don't lets collect them in this spider.

Before the yield I'd do something if storeBannerId == nofrills: brand = No Frills brand_wikidata = Q3342407 elif ...

You can find some of the wikidatas [here](https://www.wikidata.org/wiki/Q909856#P355), you may have to search for some not listed.

**Comment by boomerwv1** [comment]
Works for me. I'll update that.

File: locations/spiders/loblaws.py (hunk 2 above)
Comment (Cj-Malone) [reply locations/spiders/loblaws.py]: This seems to have been reverted, https://data.alltheplaces.xyz/map.html?show=https://data.alltheplaces.xyz/ci/3274104279/loblaws/output.geojson

**Review by Cj-Malone — APPROVED** [review]
LGTM, thank you 😊

**Pull request merged**